### PR TITLE
Fix #826 consider foreach exception only for direct children

### LINKED ISSUE #826: Regression in 2.9.0: False positive using ShortVariable for `$e` in catch() within a foreach()
- PHPMD version: 2.9.0-2.9.1
- PHP Version: 7.3.19
- Installation type: composer
- Operating System / Distribution & Version: MacOS 10.15.6 / Alpine 3.11

### Current Behavior
False positive for `ShortVariable`, regression since earlier versions:
```
$ src/vendor/bin/phpmd src/src/proof.php text phpmd.xml
src/src/proof.php:18	Avoid variables with short names like $e. Configured minimum length is 2.
```

### Expected Behavior
No errors.

### Steps To Reproduce:
Proof of concept: https://github.com/Doqnach/phpmd-shortvariable-error

The problem seems to be related to PR #807 (2.9.0). Works like expected in 2.8.2.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

### PR DESCRIPTION
**Pull request by kylekatarnls**
Type: bugfix
Issue: Resolves #826
Breaking change: no

### CODE REVIEW

File: src/main/php/PHPMD/Rule/Naming/ShortVariable.php [hunk 1]
```diff
@@ -171,7 +171,9 @@ protected function getExceptionsList()
      */
     protected function isNameAllowedInContext(AbstractNode $node)
     {
-        if ($this->isChildOf($node, 'ForeachStatement')) {
+        $parent = $node->getParent();
+
+        if ($parent && $parent->isInstanceOf('ForeachStatement')) {
```
Comment (Doqnach) [inline src/main/php/PHPMD/Rule/Naming/ShortVariable.php:176]: correct me if I'm wrong, but does this cover the following use case?:
```
foreach([1, 2] as $n) {
  if (true) {
    echo $n;
  }
}
```
and is it supposed to?

**Comment by kylekatarnls** [comment]
Switched to draft until checking variable re-use in foreach.

File: src/main/php/PHPMD/Rule/Naming/ShortVariable.php (hunk 1 above)
Comment (kylekatarnls) [reply src/main/php/PHPMD/Rule/Naming/ShortVariable.php:176]: Yes this case is fine.

**Comment by kylekatarnls** [comment]
Tested OK.

**Review by ravage84 — APPROVED** [review]

**Pull request merged**